sound/soc/sof/sof-pci-dev.c

```diff
@@ -214,6 +214,10 @@ int sof_pci_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
 		return ret;
 
 	sof_pdata->name = pci_name(pci);
+	sof_pdata->subsystem_vendor = pci->subsystem_vendor;
+	sof_pdata->subsystem_device = pci->subsystem_device;
+	sof_pdata->subsystem_id_valid = true;
```

Comment:
Not here though, this is generic code. Maybe in the machine driver. But even then only if it causes a problem.
Premature workarounds and like premature optimization - you spend a lot of time trying to fix a problem you don't have, and create new problems.
For example how do you decide it's bad? You can't assume a vendor of 8086 is bad, it could be a real unique ID on a piece of Intel hardware. And what do you do if it is bad? Ignoring it and running without any SSID would at least break the Cirrus test farm because we need an SSID of some sort to test this functionality, and on our test farm boards it doesn't matter that the UpX value is wrong, it's good enough for automated testing. It won't even cause problems for Cirrus in production code because we'd never release a firmware for that SSID.

Reply:
Fair enough, maybe adding a comment here would help. It's a bit strange that the subsystem is always valid.

Reply:
I think the problem here is my use of the word 'valid'. The flag means "a value has been written to the subsystem_* fields". Not that the value there is useful. The default 80867270 reported by the UpXtreme is a valid SSID pair, even it isn't unique.
I should perhaps call it "present" or "set" to indicate there's a value in these fields.
The problem is that normally we kzalloc() the structures and then any fields that were not written will have zero. But as far as I can see PCI doesn't not define 0 as an invalid value so it's not strictly safe to assume 0 = SSID not present